astacus/node/snapshotter.py

```diff
@@ -101,27 +101,32 @@ def _snapshotfile_from_path(self, relative_path):
         st = src_path.stat()
         return SnapshotFile(relative_path=relative_path, mtime_ns=st.st_mtime_ns, file_size=st.st_size)
 
+    def _files_are_the_same(self, old: SnapshotFile, new: SnapshotFile) -> bool:
+        return old.mtime_ns == new.mtime_ns and old.relative_path == new.relative_path and old.file_size == new.file_size
+
     def _get_snapshot_hash_list(self, relative_paths):
         same = 0
         lost = 0
         for relative_path in relative_paths:
             old_snapshotfile = self.relative_path_to_snapshotfile.get(relative_path)
             try:
-                snapshotfile = self._snapshotfile_from_path(relative_path)
+                new_snapshotfile = self._snapshotfile_from_path(relative_path)
             except FileNotFoundError:
                 lost += 1
                 if increase_worth_reporting(lost):
                     logger.debug("#%d. lost - %s disappeared before stat, ignoring", lost, self.src / relative_path)
                 continue
-            if old_snapshotfile:
-                snapshotfile.hexdigest = old_snapshotfile.hexdigest
-                snapshotfile.content_b64 = old_snapshotfile.content_b64
-                if old_snapshotfile == snapshotfile:
-                    same += 1
-                    if increase_worth_reporting(same):
-                        logger.debug("#%d. same - %r in %s is same", same, old_snapshotfile, relative_path)
-                    continue
-            yield snapshotfile
+            if old_snapshotfile and self._files_are_the_same(old_snapshotfile, new_snapshotfile):
+                new_snapshotfile.hexdigest = old_snapshotfile.hexdigest
+                new_snapshotfile.content_b64 = old_snapshotfile.content_b64
+                same += 1
+                if increase_worth_reporting(same):
+                    logger.debug("#%d. same - %r in %s is same", same, old_snapshotfile, relative_path)
+                continue
+            else:
+                new_snapshotfile = self._compute_and_set_hash(snapshotfile=new_snapshotfile)
```

Comment:
This is unnecessary. The returned snapshotfiles are changed and hashed anyway later so you're refactoring more than you need to.

Reply:
:heavy_check_mark:  thanks i removed that part and tests are still passing